app/my-bets/page.tsx

```diff
@@ -108,27 +106,19 @@ export default function MyBetsPage() {
     [userPositionsComplete]
   );
 
-  const filterUnredeemedBets = useMemo(() => {
-    return filterCompleteBets.filter(userPosition => {
-      const position = new Position(userPosition.position);
-      const outcomeIndex = position.outcomeIndex - 1;
-
-      const condition = userPosition.condition;
-      const market = condition && new Market(condition?.fixedProductMarketMakers[0]);
-      const outcomeBalance = tradesOutcomeBalance({
-        fpmmTrades: userPosition.fpmmTrades,
-      });
-      const userPositionCondition = userPosition.position.conditions[0];
-
-      const isClaimed = !outcomeBalance;
-      const isWinner = market.isWinner(outcomeIndex);
-
-      const isResolved = userPositionCondition.resolved;
-      const hasPayoutDenominator = +userPositionCondition.payoutDenominator > 0;
-
-      return isWinner && isResolved && !isClaimed && hasPayoutDenominator;
-    });
-  }, [filterCompleteBets]);
+  const filterUnredeemedBets = useMemo(
+    () =>
+      userPositionsComplete?.filter(userPosition => {
+        const marketModel = new Market(userPosition.market);
+        const position = new Position(userPosition.position);
+        const outcomeIndex = position.getOutcomeIndex();
+        const canClaim = marketModel.canClaim(outcomeIndex, position.condition);
+        const alreadyClaimed = canClaim && +userPosition.balance === 0;
+        const canRedeem = canClaim && !alreadyClaimed;
+        return canRedeem;
```

Comment:
WDYT about this:
I think this logic can be simplified, because now `canRedeem` checks if `canClaim && !(canClaim && +userPosition.balance === 0)`, but what I understand is that the user can redeem if they can claim and has balance, right?

```suggestion
        const canRedeem = canClaim && +userPosition.balance > 0;

        return canRedeem;
```

Reply:
I will keep it like this, as it will change in upcoming PR.  We'll have to support bigInt and numbers on this function.